lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -2065,7 +2065,11 @@ sub done ($self, %args) {
     if (($result_unknown || $reason_unknown) && defined $reason) {
         # restart incompletes when the reason matches the configured regex
         my $auto_clone_regex = OpenQA::App->singleton->config->{global}->{auto_clone_regex};
-        $restart = 1 if $result eq INCOMPLETE && $auto_clone_regex && $reason =~ $auto_clone_regex;
+        if ($result eq INCOMPLETE and $auto_clone_regex && $reason =~ $auto_clone_regex) {
```

Comment:
Mixing `and` and `&&` in the same line of code isn't nice. The code will do what it is supposed to do but I guess a bit more consistency would be better:

```suggestion
        if ($result eq INCOMPLETE && $auto_clone_regex && $reason =~ $auto_clone_regex) {
```